libraries/AP_HAL_SITL/SITL_cmdline.cpp

```diff
@@ -252,17 +334,21 @@ void SITL_State::_parse_command_line(int argc, char * const argv[])
         exit(1);
     }
 
-    for (uint8_t i=0; i < ARRAY_SIZE(model_constructors); i++) {
-        if (strncasecmp(model_constructors[i].name, model_str, strlen(model_constructors[i].name)) == 0) {
-            sitl_model = model_constructors[i].constructor(home_str, model_str);
-            sitl_model->set_speedup(speedup);
-            sitl_model->set_instance(_instance);
-            sitl_model->set_autotest_dir(autotest_dir);
-            _synthetic_clock_mode = true;
-            printf("Started model %s at %s at speed %.1f\n", model_str, home_str, speedup);
-            break;
+    if (simModelMap.count(model_str))
```

Comment:
this if could be improved but I don't know how. That why I came with a map ... 
The problem was to be able  to select only sim we want to  use set_interface_port . With the previous solution, I didn't find a way to do it. So I use a map, but still use the struct with constructor....